Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -1003,6 +1003,21 @@ void RenderLayerCompositor::updateEventRegions()
     m_renderView.setNeedsEventRegionUpdateForNonCompositedFrame(false);
 }
 
+EventTrackingRegions RenderLayerCompositor::getTouchEventRegions()
+{
+    EventTrackingRegions eventRegion;
+    Vector<RenderLayer*> layers = { m_renderView.layer() };
+    while (!layers.isEmpty()) {
+        if (auto* layer = layers.takeLast()) {
+            eventRegion.unite(layer->getTouchEventRegion());
```

Comment:
I'm not sure this is right; isn't this uniting regions in different coordinate systems?

Reply:
Right yeah we should collect them separately.